ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/StructConverter.java

```diff
@@ -21,26 +21,51 @@
 import com.spectralogic.ds3autogen.api.models.Ds3Type;
 import com.spectralogic.ds3autogen.c.helpers.C_TypeHelper;
 import com.spectralogic.ds3autogen.c.helpers.StructHelper;
+import com.spectralogic.ds3autogen.c.models.Request;
 import com.spectralogic.ds3autogen.c.models.Struct;
 import com.spectralogic.ds3autogen.c.models.StructMember;
+import com.spectralogic.ds3autogen.utils.ConverterUtil;
 
 import java.text.ParseException;
 
 public final class StructConverter {
     private StructConverter() {}
 
-    public static Struct toStruct(final Ds3Type ds3Type, final ImmutableSet<String> enumNames) throws ParseException {
+    public static Struct toStruct(final Ds3Type ds3Type, final ImmutableSet<String> enumNames, final ImmutableList<Request> allRequests) throws ParseException {
         final ImmutableList<StructMember> variablesList = convertDs3Elements(ds3Type.getElements(), enumNames);
+        final String responseTypeName = StructHelper.getResponseTypeName(ds3Type.getName());
         return new Struct(
-                StructHelper.getResponseTypeName(ds3Type.getName()),
-                variablesList);
+                responseTypeName,
+                convertNameToMarshall(ds3Type.getNameToMarshal()),
+                variablesList,
+                isTopLevelStruct(responseTypeName, allRequests));
     }
 
     private static ImmutableList<StructMember> convertDs3Elements(final ImmutableList<Ds3Element> elementsList, final ImmutableSet<String> enumNames) throws ParseException {
         final ImmutableList.Builder<StructMember> builder = ImmutableList.builder();
         for (final Ds3Element currentElement : elementsList) {
+            if (currentElement.getName().equals("StorageClass")) continue;  // Special case: ignore; unused - from S3ObjectApiBean
```

Comment:
Are we not including this property in the java autogen?  It seems like we should include it even if we are not doing anything with the value.